### PR TITLE
MANTA-5232 want rust-moray support for batch puts

### PR DESCRIPTION
**Pull request by rjloura**

### CODE REVIEW

File: src/objects.rs [hunk 1]
```diff
@@ -140,40 +196,6 @@ where
     }
 }
 
-// TODO: make method specific
-fn make_options(options: &MethodOptions) -> Value {
```
Comment (rjloura) [inline src/objects.rs:144]: I removed this in favor of implementing Serialize for Etag.

File: src/objects.rs [hunk 2]
```diff
@@ -27,28 +28,76 @@ pub struct MorayObject {
...
+    // experimental
+    pub fn specified_value(&self) -> Option<&String> {
+        match self {
+            Etag::Undefined | Etag::Nulled => None,
+            Etag::Specified(s) => Some(s),
+        }
+    }
+}
+
+impl Serialize for Etag {
```
Comment (rjloura) [inline src/objects.rs:53]: I added this to remove the `make_options()` semi-hack below.

File: src/objects.rs [hunk 3]
```diff
@@ -231,3 +234,156 @@ where
...
+pub struct BatchPutOp {
+    pub bucket: String,
+    pub options: MethodOptions,
+    pub key: String,
+    pub value: Value,
+}
+
+/// For now we only support Put operations
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct BatchUpdateOp {
```
Comment (rjloura) [inline src/objects.rs:260]: I added the framework for these other operations, and the ones I've tested work, but I haven't exhaustively tested them yet, so I've blocked their use on line 293 below.  If we get some help or someone needs them everything is here, they just need tests/examples.

File: src/objects.rs [hunk 4]
```diff
@@ -231,3 +234,156 @@ where
...
+    Ok(())
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use std::net::TcpListener;
+
+    #[test]
+    fn batch_unsupported_test() {
```
Comment (rjloura) [inline src/objects.rs:333]: Make sure we block unsupported operations.

File: src/objects.rs [hunk 5]
```diff
@@ -231,3 +234,156 @@ where
...
+            filter: String::from("(mydelete=filter)"),
+        }));
+
+        assert!(batch(&mut dummy_stream, &requests, &opts, |_| Ok(())).is_err());
+
+        listen_handle.join().unwrap();
+    }
+
+    #[test]
+    fn method_options_test() {
```
Comment (rjloura) [inline src/objects.rs:369]: Test the options serde round trip.

**Review by rhb2 — APPROVED** [review]

**Pull request merged**